packages/pull-signal/src/api/generate-api.ts

```diff
@@ -1,14 +1,72 @@
 import { writeFileSync } from "fs";
+import { Command, Option } from "commander";
 import { generateOpenApi } from "@ts-rest/open-api";
 import * as yaml from "js-yaml";
 import { contract } from "../contract/contract.js";
 
-export const openApiDocument = generateOpenApi(contract, {
-  info: {
-    title: "Pull signal Service API",
-    version: "1.0",
-  },
-});
+const semanticVersionRegex = /^([1-9]\d*|0)(\.(([1-9]\d*)|0)){2}$/;
 
-const fileOutputDocument = `./src/api/pull-signals_${openApiDocument.info.version}_.yaml`;
-writeFileSync(fileOutputDocument, yaml.dump(openApiDocument));
+function isValidSemanticVersion(version: string): boolean {
+  return semanticVersionRegex.test(version);
+}
+
+new Command()
+  .addOption(
+    new Option(
+      "-v, --version <string>",
+      "Pull signal API version"
+    ).makeOptionMandatory()
+  )
+  .hook("preAction", async (command) => {
+    const options = command.opts();
+
+    if (!isValidSemanticVersion(options.version)) {
+      throw new Error("Invalid version format. Please use semantic versioning");
+    }
+  })
+  .action(async (options) => {
```

Comment:
I would separate command implementation as argument of program.action() like this:  
https://github.com/pagopa/interop-be-signalhub-demo/blob/main/sh-demo-consumer-eservice-dd/src/cli.ts#L35

In this way Commander is only a caller of generateOpenApi()